v2.0.4 - Compilation Error

### Board

UM Feather S3

### Device Description

BME280 Module

### Hardware Configuration

N/A. Compilation error

### Version

2.04

### IDE Name

Arduino IDE and Arduino via vs.code

### Operating System

Linux (Mint 20.03)

### Flash frequency

N/A

### PSRAM enabled

yes

### Upload speed

N/A

### Description

Cannot compile sketch under 2.04 which is fine under 2.03.
Expected behavior is for the code to compile under 2.04 in the same way it does under 2.03.
`
In file included from /home/john/.arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32s3-elf/sys-include/sys/reent.h:503,
                 from /home/john/.arduino15/packages/esp32/hardware/esp32/2.0.4/tools/sdk/esp32s3/include/newlib/platform_include/sys/reent.h:17,
                 from /home/john/.arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32s3-elf/sys-include/stdio.h:60,
                 from /home/john/.arduino15/packages/esp32/hardware/esp32/2.0.4/cores/esp32/Arduino.h:27,
                 from /home/john/Arduino/libraries/Adafruit_IO_Arduino/src/AdafruitIO_Dashboard.h:19,
                 from /home/john/Arduino/libraries/Adafruit_IO_Arduino/src/AdafruitIO.h:19,
                 from /home/john/code/ard_OutdoorMonitor/ard_OutdoorMonitor.ino:1:
/home/john/.arduino15/packages/esp32/hardware/esp32/2.0.4/tools/sdk/esp32s3/include/newlib/platform_include/assert.h:20:10: fatal error: sdkconfig.h: No such file or directory
 #include <sdkconfig.h>
          ^~~~~~~~~~~~~
compilation terminated.
exit status 1
Error compiling for board UM FeatherS3.
`


### Sketch

```cpp
#include <AdafruitIO.h>
#include <Adafruit_BME280.h>
#include <Adafruit_NeoPixel.h>
#include <Adafruit_Sensor.h>
#include <Arduino.h>

#include "config.h"
#include <SPI.h>
#include <Wire.h>
//#include <cmath>

#define SEALEVELPRESSURE_HPA (1018.11) // Not really used

#define UPDATETIME 1000000 * 145 // 150 seconds between updates
#define TEMPOFFSET -1.0

#define LED_BUILTIN 13

#define NUMPIXELS 1
#define PIN_NEOPIXEL 40
#define PIN_LDO2_POWER 39

Adafruit_NeoPixel strip(NUMPIXELS, PIN_NEOPIXEL, NEO_GRB + NEO_KHZ800);

Adafruit_BME280 bme; // I2C

// set up the 'temperature' and 'humidity' feeds
AdafruitIO_Feed *temperature = io.feed("xxxx");
AdafruitIO_Feed *humidity = io.feed("xxxx");
AdafruitIO_Feed *pressure = io.feed("xxxx");

void setup() {
  float temp, pres, hum;

  strip.begin(); // Initialize pins for output
  strip.setBrightness(255);
  strip.show(); // Turn all LEDs off ASAP

  pinMode(PIN_LDO2_POWER, OUTPUT);
  digitalWrite(PIN_LDO2_POWER, HIGH);

  if (!bme.begin(0x76)) {
    strip.setPixelColor(0, 0x00FF0000);
    strip.show();
    while (1)
      ;
  }

  bme.setTemperatureCompensation(TEMPOFFSET);

  io.connect();

  // wait for a connection
  while (io.status() < AIO_CONNECTED) {
  }
  if (io.status() >= AIO_NET_CONNECTED) {
    io.run();

    temp = bme.readTemperature();
    hum = bme.readHumidity();
    pres = bme.readPressure() / 100;

    temperature->save(temp);
    humidity->save(hum);
    pressure->save(pres);

    strip.setPixelColor(0, 0x0000FF00);
    strip.show();
    delay(50); // wait for a second
    strip.clear();
    strip.show();

    digitalWrite(PIN_LDO2_POWER, LOW);

    esp_sleep_enable_timer_wakeup(UPDATETIME);
    esp_deep_sleep_start();
  }

  else {
    esp_restart();
  }
}
```


### Debug Message

```plain
N/A
```


### Other Steps to Reproduce

It compiles under 2.03.

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

> @SuGlider This seems to be isolated to ESP32-S3 (possibly with PSRAM). I'm running into the same error when compiling Adafruit WipperSnapper firmware for the **Feather ESP32-S3**. It is able to compile with QT Py S3, no PSRAM (https://github.com/brentru/Adafruit_Wippersnapper_Arduino/runs/7255321796?check_suite_focus=true#step:8:47)
> 
> ```
>   In file included from /home/runner/.arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32s3-elf/sys-include/sys/reent.h:503,
>                    from /home/runner/.arduino15/packages/esp32/hardware/esp32/2.0.4/tools/sdk/esp32s3/include/newlib/platform_include/sys/reent.h:17,
>                    from /home/runner/.arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32s3-elf/sys-include/stdio.h:60,
>                    from /home/runner/.arduino15/packages/esp32/hardware/esp32/2.0.4/cores/esp32/Arduino.h:27,
>                    from /tmp/arduino-sketch-C5C4B37D41FC5188187C73B78CE4456F/sketch/Wippersnapper_demo.ino.cpp:1:
>   /home/runner/.arduino15/packages/esp32/hardware/esp32/2.0.4/tools/sdk/esp32s3/include/newlib/platform_include/assert.h:20:10: fatal error: sdkconfig.h: No such file or directory
>    #include <sdkconfig.h>
>             ^~~~~~~~~~~~~
>   compilation terminated.
> ```
> 
> I do not have issues with the ESP32, ESP32-S2 boards in the same build matrix.
> 
> (run log: https://github.com/brentru/Adafruit_Wippersnapper_Arduino/runs/7255231513?check_suite_focus=true#logs)

Can confirm that compilation for UM Feather S2 with same sketch is just fine.

tinys3:
In file included from c:\users\hneary\appdata\local\arduino15\packages\esp32\tools\xtensa-esp32s3-elf-gcc\gcc8_4_0-esp-2021r2-patch3\xtensa-esp32s3-elf\sys-include\sys\reent.h:503,
                 from C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4/tools/sdk/esp32s3/include/newlib/platform_include/sys/reent.h:17,
                 from c:\users\hneary\appdata\local\arduino15\packages\esp32\tools\xtensa-esp32s3-elf-gcc\gcc8_4_0-esp-2021r2-patch3\xtensa-esp32s3-elf\sys-include\stdio.h:60,
                 from C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\cores\esp32/Arduino.h:27,
                 from sketch\SerialToSerialBT.ino.cpp:1:
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4/tools/sdk/esp32s3/include/newlib/platform_include/assert.h:20:10: fatal error: sdkconfig.h: No such file or directory
 #include <sdkconfig.h>
          ^~~~~~~~~~~~~
compilation terminated.
exit status 1
Error compiling for board UM TinyS3.

@SuGlider this is what happened:

Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\ledController-main
Invalid library found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main: no headers files (.h) found in C:\Users\hneary\Documents\Arduino\libraries\RejsaCAN-ESP32-main










c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL8_stop_btv+0x18): undefined reference to `esp_spp_disconnect'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL8_stop_btv+0x1c): undefined reference to `esp_spp_deinit'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL16_spp_send_bufferv+0xc): undefined reference to `esp_spp_write'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL8btSetPinv+0xc): undefined reference to `esp_bt_gap_set_pin'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZN15BluetoothSerial5beginE6Stringb+0x38): undefined reference to `esp_bt_gap_register_callback'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZN15BluetoothSerial5beginE6Stringb+0x3c): undefined reference to `esp_spp_register_callback'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZN15BluetoothSerial5beginE6Stringb+0x40): undefined reference to `esp_spp_init'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZN15BluetoothSerial5beginE6Stringb+0x44): undefined reference to `esp_bt_gap_set_security_param'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZN15BluetoothSerial5beginE6Stringb+0x48): undefined reference to `esp_bt_gap_set_cod'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL10esp_spp_cb18esp_spp_cb_event_tP18esp_spp_cb_param_t+0x28): undefined reference to `esp_bt_gap_set_scan_mode'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL10esp_spp_cb18esp_spp_cb_event_tP18esp_spp_cb_param_t+0x2c): undefined reference to `esp_spp_start_srv'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL10esp_spp_cb18esp_spp_cb_event_tP18esp_spp_cb_param_t+0x30): undefined reference to `esp_spp_connect'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL13esp_bt_gap_cb21esp_bt_gap_cb_event_tP21esp_bt_gap_cb_param_t+0x20): undefined reference to `esp_bt_gap_resolve_eir_data'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL13esp_bt_gap_cb21esp_bt_gap_cb_event_tP21esp_bt_gap_cb_param_t+0x28): undefined reference to `esp_bt_gap_cancel_discovery'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL13esp_bt_gap_cb21esp_bt_gap_cb_event_tP21esp_bt_gap_cb_param_t+0x2c): undefined reference to `esp_spp_start_discovery'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL13esp_bt_gap_cb21esp_bt_gap_cb_event_tP21esp_bt_gap_cb_param_t+0x3c): undefined reference to `esp_bt_gap_pin_reply'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o:(.literal._ZL13esp_bt_gap_cb21esp_bt_gap_cb_event_tP21esp_bt_gap_cb_param_t+0x40): undefined reference to `esp_bt_gap_ssp_confirm_reply'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `_stop_bt()':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:716: undefined reference to `esp_spp_disconnect'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:717: undefined reference to `esp_spp_deinit'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `_spp_send_buffer()':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:189: undefined reference to `esp_spp_write'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `btSetPin()':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:153: undefined reference to `esp_bt_gap_set_pin'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `BluetoothSerial::begin(String, bool)':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:651: undefined reference to `esp_bt_gap_register_callback'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `_init_bt':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:666: undefined reference to `esp_spp_register_callback'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:666: undefined reference to `esp_spp_init'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:693: undefined reference to `esp_bt_gap_set_security_param'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:697: undefined reference to `esp_bt_gap_set_cod'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `esp_spp_cb(esp_spp_cb_event_t, esp_spp_cb_param_t*)':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:265: undefined reference to `esp_bt_gap_set_scan_mode'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:271: undefined reference to `esp_spp_start_srv'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:284: undefined reference to `esp_spp_disconnect'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:364: undefined reference to `esp_spp_connect'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:393: undefined reference to `esp_spp_disconnect'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `esp_bt_gap_cb(esp_bt_gap_cb_event_t, esp_bt_gap_cb_param_t*)':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:128: undefined reference to `esp_bt_gap_resolve_eir_data'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `get_name_from_eir':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:130: undefined reference to `esp_bt_gap_resolve_eir_data'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: libraries\BluetoothSerial\BluetoothSerial.cpp.o: in function `esp_bt_gap_cb(esp_bt_gap_cb_event_t, esp_bt_gap_cb_param_t*)':
C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:457: undefined reference to `esp_bt_gap_cancel_discovery'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:458: undefined reference to `esp_spp_start_discovery'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:547: undefined reference to `esp_bt_gap_pin_reply'
c:/users/hneary/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s3-elf-gcc/gcc8_4_0-esp-2021r2-patch3/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld.exe: C:\Users\hneary\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\BluetoothSerial\src/BluetoothSerial.cpp:558: undefined reference to `esp_bt_gap_ssp_confirm_reply'
collect2.exe: error: ld returned 1 exit status
exit status 1
Error compiling for board UM TinyS3.

--- a/boards.txt
+++ b/boards.txt
@@ -65,9 +65,8 @@ esp32s3.build.partitions=default
 esp32s3.build.defines=
 esp32s3.build.loop_core=
 esp32s3.build.event_core=
-esp32s3.build.flash_type=qspi
 esp32s3.build.psram_type=qspi
-esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
+esp32s3.build.memory_type={build.boot}_{build.psram_type}
 
 esp32s3.menu.PSRAM.disabled=Disabled
 esp32s3.menu.PSRAM.disabled.build.defines=
@@ -84,25 +83,21 @@ esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 esp32s3.menu.FlashMode.qio.build.boot=qio
 esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-esp32s3.menu.FlashMode.qio.build.flash_type=qspi
 esp32s3.menu.FlashMode.qio120=QIO 120MHz
 esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 esp32s3.menu.FlashMode.qio120.build.boot=qio
 esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
 esp32s3.menu.FlashMode.dio=DIO 80MHz
 esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 esp32s3.menu.FlashMode.dio.build.boot=dio
 esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-esp32s3.menu.FlashMode.dio.build.flash_type=qspi
 esp32s3.menu.FlashMode.opi=OPI 80MHz
 esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 esp32s3.menu.FlashMode.opi.build.boot=opi
 esp32s3.menu.FlashMode.opi.build.boot_freq=80m
 esp32s3.menu.FlashMode.opi.build.flash_freq=80m
-esp32s3.menu.FlashMode.opi.build.flash_type=opi
 
 esp32s3.menu.FlashSize.4M=4MB (32Mb)
 esp32s3.menu.FlashSize.4M.build.flash_size=4MB
@@ -1101,7 +1096,7 @@ esp32s3box.build.flash_mode=dio
 esp32s3box.build.boot=qio
 esp32s3box.build.partitions=default
 esp32s3box.build.defines=-DBOARD_HAS_PSRAM
-esp32s3box.build.memory_type=qspi_opi
+esp32s3box.build.memory_type=qio_opi
 esp32s3box.build.loop_core=-DARDUINO_RUNNING_CORE=1
 esp32s3box.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
 
@@ -1212,7 +1207,7 @@ esp32s3usbotg.build.flash_mode=dio
 esp32s3usbotg.build.boot=qio
 esp32s3usbotg.build.partitions=default
 esp32s3usbotg.build.defines=
-esp32s3usbotg.build.memory_type=qspi_qspi
+esp32s3usbotg.build.memory_type=qio_qspi
 esp32s3usbotg.build.loop_core=-DARDUINO_RUNNING_CORE=1
 esp32s3usbotg.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
 
@@ -2295,9 +2290,9 @@ tinys3.build.partitions=default
 tinys3.build.defines=
 tinys3.build.loop_core=
 tinys3.build.event_core=
-tinys3.build.flash_type=qspi
+tinys3.build.flash_type=qio
 tinys3.build.psram_type=qspi
-tinys3.build.memory_type=qspi_qspi
+tinys3.build.memory_type=qio_qspi
 
 tinys3.menu.LoopCore.1=Core 1
 tinys3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -2442,9 +2437,9 @@ pros3.build.partitions=default
 pros3.build.defines=
 pros3.build.loop_core=
 pros3.build.event_core=
-pros3.build.flash_type=qspi
+pros3.build.flash_type=qio
 pros3.build.psram_type=qspi
-pros3.build.memory_type=qspi_qspi
+pros3.build.memory_type=qio_qspi
 
 pros3.menu.LoopCore.1=Core 1
 pros3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -2598,9 +2593,9 @@ feathers3.build.partitions=default
 feathers3.build.defines=
 feathers3.build.loop_core=
 feathers3.build.event_core=
-feathers3.build.flash_type=qspi
+feathers3.build.flash_type=qio
 feathers3.build.psram_type=qspi
-feathers3.build.memory_type=qspi_qspi
+feathers3.build.memory_type=qio_qspi
 
 feathers3.menu.LoopCore.1=Core 1
 feathers3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -5908,7 +5903,7 @@ dfrobot_firebeetle2_esp32s3.build.partitions=default
 dfrobot_firebeetle2_esp32s3.build.defines=
 dfrobot_firebeetle2_esp32s3.build.loop_core=
 dfrobot_firebeetle2_esp32s3.build.event_core=
-dfrobot_firebeetle2_esp32s3.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.build.psram_type=qspi
 dfrobot_firebeetle2_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
@@ -5927,19 +5922,19 @@ dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.boot=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120=QIO 120MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.boot=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio=DIO 80MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.boot=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi=OPI 80MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi.build.boot=opi
@@ -7717,7 +7712,7 @@ adafruit_feather_esp32s3.build.partitions=default
 adafruit_feather_esp32s3.build.defines=
 adafruit_feather_esp32s3.build.loop_core=
 adafruit_feather_esp32s3.build.event_core=
-adafruit_feather_esp32s3.build.flash_type=qspi
+adafruit_feather_esp32s3.build.flash_type=qio
 adafruit_feather_esp32s3.build.psram_type=qspi
 adafruit_feather_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
@@ -7726,19 +7721,19 @@ adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.boot=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_type=qspi
+adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_type=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio120=QIO 120MHz
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.boot=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
+adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_type=qio
 adafruit_feather_esp32s3.menu.FlashMode.dio=DIO 80MHz
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.boot=dio
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_type=qspi
+adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_type=qio
 adafruit_feather_esp32s3.menu.FlashMode.opi=OPI 80MHz
 adafruit_feather_esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 adafruit_feather_esp32s3.menu.FlashMode.opi.build.boot=opi

--- a/boards.txt
+++ b/boards.txt
@@ -1096,7 +1096,7 @@ esp32s3box.build.flash_mode=dio
 esp32s3box.build.boot=qio
 esp32s3box.build.partitions=default
 esp32s3box.build.defines=-DBOARD_HAS_PSRAM
-esp32s3box.build.memory_type=qspi_opi
+esp32s3box.build.memory_type=qio_opi
 esp32s3box.build.loop_core=-DARDUINO_RUNNING_CORE=1
 esp32s3box.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
 
@@ -1207,7 +1207,7 @@ esp32s3usbotg.build.flash_mode=dio
 esp32s3usbotg.build.boot=qio
 esp32s3usbotg.build.partitions=default
 esp32s3usbotg.build.defines=
-esp32s3usbotg.build.memory_type=qspi_qspi
+esp32s3usbotg.build.memory_type=qio_qspi
 esp32s3usbotg.build.loop_core=-DARDUINO_RUNNING_CORE=1
 esp32s3usbotg.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
 
@@ -2290,9 +2290,9 @@ tinys3.build.partitions=default
 tinys3.build.defines=
 tinys3.build.loop_core=
 tinys3.build.event_core=
-tinys3.build.flash_type=qspi
+tinys3.build.flash_type=qio
 tinys3.build.psram_type=qspi
-tinys3.build.memory_type=qspi_qspi
+tinys3.build.memory_type=qio_qspi
 
 tinys3.menu.LoopCore.1=Core 1
 tinys3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -2437,9 +2437,9 @@ pros3.build.partitions=default
 pros3.build.defines=
 pros3.build.loop_core=
 pros3.build.event_core=
-pros3.build.flash_type=qspi
+pros3.build.flash_type=qio
 pros3.build.psram_type=qspi
-pros3.build.memory_type=qspi_qspi
+pros3.build.memory_type=qio_qspi
 
 pros3.menu.LoopCore.1=Core 1
 pros3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -2593,9 +2593,9 @@ feathers3.build.partitions=default
 feathers3.build.defines=
 feathers3.build.loop_core=
 feathers3.build.event_core=
-feathers3.build.flash_type=qspi
+feathers3.build.flash_type=qio
 feathers3.build.psram_type=qspi
-feathers3.build.memory_type=qspi_qspi
+feathers3.build.memory_type=qio_qspi
 
 feathers3.menu.LoopCore.1=Core 1
 feathers3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -5903,7 +5903,7 @@ dfrobot_firebeetle2_esp32s3.build.partitions=default
 dfrobot_firebeetle2_esp32s3.build.defines=
 dfrobot_firebeetle2_esp32s3.build.loop_core=
 dfrobot_firebeetle2_esp32s3.build.event_core=
-dfrobot_firebeetle2_esp32s3.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.build.psram_type=qspi
 dfrobot_firebeetle2_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
@@ -5922,19 +5922,19 @@ dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.boot=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120=QIO 120MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.boot=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio=DIO 80MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.boot=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_type=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi=OPI 80MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi.build.boot=opi
@@ -7712,7 +7712,7 @@ adafruit_feather_esp32s3.build.partitions=default
 adafruit_feather_esp32s3.build.defines=
 adafruit_feather_esp32s3.build.loop_core=
 adafruit_feather_esp32s3.build.event_core=
-adafruit_feather_esp32s3.build.flash_type=qspi
+adafruit_feather_esp32s3.build.flash_type=qio
 adafruit_feather_esp32s3.build.psram_type=qspi
 adafruit_feather_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
@@ -7721,19 +7721,19 @@ adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.boot=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_type=qspi
+adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_type=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio120=QIO 120MHz
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.boot=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
+adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_type=qio
 adafruit_feather_esp32s3.menu.FlashMode.dio=DIO 80MHz
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.boot=dio
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_type=qspi
+adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_type=qio
 adafruit_feather_esp32s3.menu.FlashMode.opi=OPI 80MHz
 adafruit_feather_esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 adafruit_feather_esp32s3.menu.FlashMode.opi.build.boot=opi

--- a/boards.txt
+++ b/boards.txt
@@ -65,8 +65,9 @@ esp32s3.build.partitions=default
 esp32s3.build.defines=
 esp32s3.build.loop_core=
 esp32s3.build.event_core=
+esp32s3.build.flash_type=qspi
 esp32s3.build.psram_type=qspi
-esp32s3.build.memory_type={build.boot}_{build.psram_type}
+esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
 esp32s3.menu.PSRAM.disabled=Disabled
 esp32s3.menu.PSRAM.disabled.build.defines=
@@ -83,21 +84,25 @@ esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 esp32s3.menu.FlashMode.qio.build.boot=qio
 esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 esp32s3.menu.FlashMode.qio.build.flash_freq=80m
+esp32s3.menu.FlashMode.qio.build.flash_type=qspi
 esp32s3.menu.FlashMode.qio120=QIO 120MHz
 esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 esp32s3.menu.FlashMode.qio120.build.boot=qio
 esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
+esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
 esp32s3.menu.FlashMode.dio=DIO 80MHz
 esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 esp32s3.menu.FlashMode.dio.build.boot=dio
 esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 esp32s3.menu.FlashMode.dio.build.flash_freq=80m
+esp32s3.menu.FlashMode.dio.build.flash_type=qspi
 esp32s3.menu.FlashMode.opi=OPI 80MHz
 esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 esp32s3.menu.FlashMode.opi.build.boot=opi
 esp32s3.menu.FlashMode.opi.build.boot_freq=80m
 esp32s3.menu.FlashMode.opi.build.flash_freq=80m
+esp32s3.menu.FlashMode.opi.build.flash_type=opi
 
 esp32s3.menu.FlashSize.4M=4MB (32Mb)
 esp32s3.menu.FlashSize.4M.build.flash_size=4MB
@@ -1096,7 +1101,7 @@ esp32s3box.build.flash_mode=dio
 esp32s3box.build.boot=qio
 esp32s3box.build.partitions=default
 esp32s3box.build.defines=-DBOARD_HAS_PSRAM
-esp32s3box.build.memory_type=qio_opi
+esp32s3box.build.memory_type=qspi_opi
 esp32s3box.build.loop_core=-DARDUINO_RUNNING_CORE=1
 esp32s3box.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
 
@@ -1207,7 +1212,7 @@ esp32s3usbotg.build.flash_mode=dio
 esp32s3usbotg.build.boot=qio
 esp32s3usbotg.build.partitions=default
 esp32s3usbotg.build.defines=
-esp32s3usbotg.build.memory_type=qio_qspi
+esp32s3usbotg.build.memory_type=qspi_qspi
 esp32s3usbotg.build.loop_core=-DARDUINO_RUNNING_CORE=1
 esp32s3usbotg.build.event_core=-DARDUINO_EVENT_RUNNING_CORE=1
 
@@ -2290,9 +2295,9 @@ tinys3.build.partitions=default
 tinys3.build.defines=
 tinys3.build.loop_core=
 tinys3.build.event_core=
-tinys3.build.flash_type=qio
+tinys3.build.flash_type=qspi
 tinys3.build.psram_type=qspi
-tinys3.build.memory_type=qio_qspi
+tinys3.build.memory_type=qspi_qspi
 
 tinys3.menu.LoopCore.1=Core 1
 tinys3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -2437,9 +2442,9 @@ pros3.build.partitions=default
 pros3.build.defines=
 pros3.build.loop_core=
 pros3.build.event_core=
-pros3.build.flash_type=qio
+pros3.build.flash_type=qspi
 pros3.build.psram_type=qspi
-pros3.build.memory_type=qio_qspi
+pros3.build.memory_type=qspi_qspi
 
 pros3.menu.LoopCore.1=Core 1
 pros3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -2593,9 +2598,9 @@ feathers3.build.partitions=default
 feathers3.build.defines=
 feathers3.build.loop_core=
 feathers3.build.event_core=
-feathers3.build.flash_type=qio
+feathers3.build.flash_type=qspi
 feathers3.build.psram_type=qspi
-feathers3.build.memory_type=qio_qspi
+feathers3.build.memory_type=qspi_qspi
 
 feathers3.menu.LoopCore.1=Core 1
 feathers3.menu.LoopCore.1.build.loop_core=-DARDUINO_RUNNING_CORE=1
@@ -5903,7 +5908,7 @@ dfrobot_firebeetle2_esp32s3.build.partitions=default
 dfrobot_firebeetle2_esp32s3.build.defines=
 dfrobot_firebeetle2_esp32s3.build.loop_core=
 dfrobot_firebeetle2_esp32s3.build.event_core=
-dfrobot_firebeetle2_esp32s3.build.flash_type=qio
+dfrobot_firebeetle2_esp32s3.build.flash_type=qspi
 dfrobot_firebeetle2_esp32s3.build.psram_type=qspi
 dfrobot_firebeetle2_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
@@ -5922,19 +5927,19 @@ dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.boot=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_type=qio
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio.build.flash_type=qspi
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120=QIO 120MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.boot=qio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_type=qio
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio=DIO 80MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.boot=dio
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_type=qio
+dfrobot_firebeetle2_esp32s3.menu.FlashMode.dio.build.flash_type=qspi
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi=OPI 80MHz
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 dfrobot_firebeetle2_esp32s3.menu.FlashMode.opi.build.boot=opi
@@ -7712,7 +7717,7 @@ adafruit_feather_esp32s3.build.partitions=default
 adafruit_feather_esp32s3.build.defines=
 adafruit_feather_esp32s3.build.loop_core=
 adafruit_feather_esp32s3.build.event_core=
-adafruit_feather_esp32s3.build.flash_type=qio
+adafruit_feather_esp32s3.build.flash_type=qspi
 adafruit_feather_esp32s3.build.psram_type=qspi
 adafruit_feather_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
 
@@ -7721,19 +7726,19 @@ adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.boot=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.boot_freq=80m
 adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_type=qio
+adafruit_feather_esp32s3.menu.FlashMode.qio.build.flash_type=qspi
 adafruit_feather_esp32s3.menu.FlashMode.qio120=QIO 120MHz
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.boot=qio
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.boot_freq=120m
 adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_type=qio
+adafruit_feather_esp32s3.menu.FlashMode.qio120.build.flash_type=qspi
 adafruit_feather_esp32s3.menu.FlashMode.dio=DIO 80MHz
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_mode=dio
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.boot=dio
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.boot_freq=80m
 adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_freq=80m
-adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_type=qio
+adafruit_feather_esp32s3.menu.FlashMode.dio.build.flash_type=qspi
 adafruit_feather_esp32s3.menu.FlashMode.opi=OPI 80MHz
 adafruit_feather_esp32s3.menu.FlashMode.opi.build.flash_mode=dout
 adafruit_feather_esp32s3.menu.FlashMode.opi.build.boot=opi

